### PR TITLE
Fix boundary condition bug

### PR DESCRIPTION
**Pull request by jamiebramwell**
This fixes a bug in the boundary condition class where the correct degrees of freedom were not being calculated properly for vector-valued functions.

### CODE REVIEW

**Review by white238 — APPROVED** [review]

File: src/serac/physics/tests/serac_solid_functional_boundary.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Perform the quasi-static solve
+  double dt = 0.1;
+
+  solid_solver.advanceTimestep(dt);
+  solid_solver.outputState("paraview_output");
+
+  // Check the final displacement norm
+  EXPECT_NEAR(0.171720048, norm(solid_solver.displacement()), 1.0e-6);
+
```
Comment (samuelpmishLLNL) [inline src/serac/physics/tests/serac_solid_functional_boundary.cpp]: I don't want to be too much of a jerk here, but I said earlier that I will not approve any PRs that introduce tests like this. I understand this is a 1-line-of-code change PR, but we can't keep doing this.

Let's start doing basic patch tests or something-- linear elasticity is fine for testing this feature, so we have analytic solutions to compare the entire displacement solution to.

I just glanced at mfem and didn't see any obvious patch test meshes in their repo (?!) so I guess we have to write our own, but it's pretty easy to do.

Comment (jamiebramwell) [reply src/serac/physics/tests/serac_solid_functional_boundary.cpp]: I totally agree. I tried to be a bit better here by actually checking the exact dof values. I agree we need real patch tests.

Comment (white238) [reply src/serac/physics/tests/serac_solid_functional_boundary.cpp]: I created an issue so we can track this and start the conversation outside of a bugfix PR.

#754

**Review by samuelpmishLLNL — APPROVED** [review]

File: src/serac/physics/tests/serac_solid_functional_boundary.cpp [hunk 2]
```diff
@@ -80,8 +84,18 @@ TEST(SolidFunctional, BoundaryCondition)
   solid_solver.advanceTimestep(dt);
   solid_solver.outputState("paraview_output");
 
-  // Check the final displacement norm
-  EXPECT_NEAR(0.171720048, norm(solid_solver.displacement()), 1.0e-6);
+  // compare approximate solution to the exact,
```
Comment (jamiebramwell) [inline src/serac/physics/tests/serac_solid_functional_boundary.cpp:87]: This is great! Thanks!!!

**Pull request merged**